horovod/common/operations.cc

```diff
@@ -1109,6 +1158,37 @@ void BackgroundThreadLoop(HorovodGlobalState& state, MPIContext& ctx) {
     state.message_table = std::unique_ptr<MessageTable>(new MessageTable());
   }
 
+  state.cpu_operation = HOROVOD_MPI;
+
+#if HAVE_GLOO
+  state.cpu_operation = HOROVOD_GLOO;
+#endif
+
+#if HAVE_MLSL
+  state.cpu_operation = HOROVOD_MLSL;
+#endif
+
+  // If specified by admin during compiling
+#if HOROVOD_CPU_OPERATIONS
+  state.cpu_operation = HOROVOD_CPU_OPERATIONS;
+#endif
+
+  // If specified by user during runtime
+  auto user_cpu_operation = std::getenv(HOROVOD_CPU_RUNTIME);
```

Comment:
I think it's fine to use the same environment variable for compile time and runtime, `HOROVOD_CPU_OPERATIONS`.

Reply:
That would conflict with the previous compilation flag `#if HOROVOD_CPU_OPERATIONS`. The compiler will throw an error.

Reply:
I see, you kept the environment variable the same, so this is fine.